applications/HDF5Application/custom_processes/hdf5_xdmf_connectivities_writer_process.cpp

```diff
@@ -0,0 +1,111 @@
+#include "custom_processes/hdf5_xdmf_connectivities_writer_process.h"
+
+#include "custom_io/hdf5_file_serial.h"
+#include "utilities/openmp_utils.h"
+
+namespace Kratos
+{
+namespace HDF5
+{
+XdmfConnectivitiesWriterProcess::XdmfConnectivitiesWriterProcess(const std::string& rFileName, const std::string& rPrefix)
+{
+    KRATOS_TRY;
+
+    Parameters file_params(R"(
+            {
+                "file_name" : "",
+                "file_access_mode": "read_write"
+            })");
+    file_params["file_name"].SetString(rFileName);
+    mpFile = FileSerial::Pointer(new FileSerial(file_params));
+    mPrefix = rPrefix;
+    std::string node_ids_path = mPrefix + "/Nodes/Local/Ids";
+
+    KRATOS_ERROR_IF(mpFile->HasPath(node_ids_path) == false)
+        << "Path \"" << node_ids_path << "\" was not found." << std::endl;
+
+    Vector<int> node_ids;
+    const unsigned num_points = mpFile->GetDataDimensions(node_ids_path)[0];
+    mpFile->ReadDataSet(node_ids_path, node_ids, 0, num_points);
+
+    // Set the parametric coordinate ids.
+    mKratosToXdmfIdTable.resize(num_points);
+    const int num_threads = OpenMPUtils::GetNumThreads();
+    OpenMPUtils::PartitionVector partition;
+    OpenMPUtils::DivideInPartitions(num_points, num_threads, partition);
+#pragma omp parallel
+    {
+        const int thread_id = OpenMPUtils::ThisThread();
+        for (auto i = partition[thread_id]; i < partition[thread_id + 1]; ++i)
+        {
+            // Here we expect the ids are one-based indices.
+            mKratosToXdmfIdTable(node_ids[i] - 1) = i;
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+void XdmfConnectivitiesWriterProcess::Execute()
+{
+    KRATOS_TRY;
+
+    std::vector<std::string> labels;
+    Matrix<int> connectivities;
```

Comment:
`Matrix` or `std::vector<std::vector<int>>`?

Reply:
Matrix

Reply:
actually it appears to be unused so neither :+1: